cmake/local_template.cmake

```diff
@@ -0,0 +1,4 @@
+# set(ZEPHYR_SDK_INSTALL_DIR $ENV{HOME}/zephyr-sdk)
+# set(ZEPHYR_TOOLCHAIN_VARIANT zephyr)
```

Comment:
and how do you switch between variants, this is something I do on daily basis.

Reply:
With bash I have 

sebo@mach:~$ ls .zephyr*
.zephyr_clang.sh          .zephyr_esp.sh        .zephyr_issm.sh    .zephyrrc           .zephyr_sdk.sh  .zephyr_xtools.sh
.zephyr_cross_compile.sh  .zephyr_gccarmemb.sh  .zephyr_kbuild.sh  .zephyr_riscv32.sh  .zephyr_xcc.sh

today to switch between variants.

with the new scheme you could switch between variants in a session like this:


local.cmake:
```
if($ENV{zt} STREQUAL gccarmemb)
  set(GCCARMEMB_TOOLCHAIN_PATH /home/sebo/Downloads/gcc-arm-none-eabi-7-2018-q2-update/)
  set(ZEPHYR_TOOLCHAIN_VARIANT gccarmemb)
else()
  set(ZEPHYR_SDK_INSTALL_DIR $ENV{HOME}/zephyr-sdk)
  set(ZEPHYR_TOOLCHAIN_VARIANT zephyr)
endif()

set_ifndef(BOARD nrf52_pca100401)
```

```
sebo@mach:~$ export zt=gccarmemb
sebo@mach:~$ cmake ..
```

Reply:
> With bash I have
> 
> sebo@mach:~$ ls .zephyr*
> .zephyr_clang.sh .zephyr_esp.sh .zephyr_issm.sh .zephyrrc .zephyr_sdk.sh .zephyr_xtools.sh
> .zephyr_cross_compile.sh .zephyr_gccarmemb.sh .zephyr_kbuild.sh .zephyr_riscv32.sh .zephyr_xcc.sh

and all I have to do is just set the correct variant and maintain all related env variables in ~/.zephyrrc

this will also most likely break how sanitycheck works btw.

Reply:
> I wish to deprecate and discourage the use of zephyr-env.sh

why? All that does is set ZEPHYR_BASE, you will have to export the variable using some other mean and the argument of forgetting to do that will still apply, i.e. someone might forget to set ZEPHYR_BASE in the new terminal.

sanitycheck does not depend on zephyr-env.sh btw, but it does depend on the environment variables.

Reply:
Because there are fewer usability issues[0] when users set ZEPHYR_BASE and maintain a local.cmake file than when they run zephyr-env.cmd and maintain a .zephyrrc file.

[0] The three usability issues are described in the commit message. True, they are not resolved for ZEPHYR_BASE, but fewer env vars means fewer user-error opportunities.